src/api/entities/Identity/index.ts

```diff
@@ -742,13 +759,61 @@ export class Identity extends Entity<UniqueIdentifiers, string> {
 
     const rawPermissions = await identity.keyRecords.multi(identityKeys);
 
-    return assembleResult(rawPermissions, identityKeys);
+    return {
+      data: assembleResult(rawPermissions, identityKeys),
+      next,
+    };
+  }
+
+  /**
+   * Fetches account permissions for the given secondary Accounts
+   *
+   * @throws if given an Account that belongs to a different Identity. Unlinked Accounts are ignored
+   */
+  public async getSecondaryAccountPermissions(args: {
+    accounts: Account[];
+  }): Promise<PermissionedAccount[]> {
+    const {
+      did,
+      context,
+      context: {
+        polymeshApi: {
+          query: { identity },
+        },
+      },
+    } = this;
+
+    const { accounts } = args;
+
+    const identityKeys = accounts.map(a => stringToAccountId(a.address, context));
+    const results = await identity.keyRecords.multi(identityKeys);
+    return results.reduce((result: PermissionedAccount[], optKeyRecord, index) => {
+      const account = accounts[index];
+      const record = optKeyRecord.unwrap();
+      if (record.isSecondaryKey) {
+        const [rawIdentityId, rawPermissions] = record.asSecondaryKey;
+
+        if (identityIdToString(rawIdentityId) !== did) {
+          throw new PolymeshError({
+            code: ErrorCode.General,
+            message: 'Given an account that belongs to another Identity',
+          });
+        }
```

Comment:
Remember to remove this when you transform the method into a util

Reply:
Good idea about having getPermisionedAccounts calling this. I made it support a callback, it doesn't take pagination params though since its a wrapper around the `.multi` call. `getSecondaryAccounts` will still handle pagination part.